packages/fluentui/react-northstar/src/components/Input/Input.tsx

```diff
@@ -78,161 +92,214 @@ export interface InputProps extends UIComponentProps, ChildrenComponentProps, Su
-    const [htmlInputProps, restProps] = partitionHTMLProps(unhandledProps);
+  const computeIcon = (): ShorthandValue<BoxProps> => {
+    const { clearable, icon } = props;
 
-    return Box.create(wrapper, {
-      defaultProps: () => ({
-        ...accessibility.attributes.root,
+    if (clearable && (value as string)?.length !== 0) {
+      return {};
+    }
+
+    return icon || null;
+  };
+
+  const element = Box.create(wrapper, {
+    defaultProps: () =>
+      getA11yProps('root', {
         className: cx(inputClassName, className),
         children: (
           <>
             <Ref
               innerRef={(inputElement: HTMLElement) => {
-                handleRef(this.inputRef, inputElement);
                 handleRef(inputRef, inputElement);
+                handleRef(props.inputRef, inputElement);
               }}
             >
               {Box.create(input || type, {
-                defaultProps: () => ({
-                  ...htmlInputProps,
-                  as: 'input',
-                  disabled,
-                  type,
-                  value,
-                  className: inputSlotClassNames.input,
-                  styles: styles.input,
-                  onChange: this.handleChange,
-                  ...applyAccessibilityKeyHandlers(accessibility.keyHandlers.input, htmlInputProps),
-                }),
+                defaultProps: () =>
+                  getA11yProps('input', {
+                    ...htmlInputProps,
+                    as: 'input',
+                    disabled,
+                    type,
+                    value: value || '',
+                    className: inputSlotClassNames.input,
+                    styles: resolvedStyles.input,
+                    onChange: handleChange,
+                  }),
               })}
             </Ref>
-            {Box.create(this.computeIcon(), {
-              defaultProps: () => ({
-                className: inputSlotClassNames.icon,
-                styles: styles.icon,
-              }),
-              overrideProps: this.handleIconOverrides,
+            {Box.create(computeIcon(), {
+              defaultProps: () =>
+                getA11yProps('icon', {
+                  className: inputSlotClassNames.icon,
+                  styles: resolvedStyles.icon,
+                }),
+              overrideProps: handleIconOverrides,
             })}
           </>
         ),
-        styles: styles.root,
+        styles: resolvedStyles.root,
         ...restProps,
       }),
-      overrideProps: {
-        as: (wrapper && (wrapper as any).as) || ElementType,
-      },
-    });
-  }
-
-  handleIconOverrides = predefinedProps => ({
-    onClick: (e: React.SyntheticEvent) => {
-      if (!this.props.disabled) {
-        this.handleOnClear(e);
-        this.inputRef.current.focus();
-      }
-
-      _.invoke(predefinedProps, 'onClick', e, this.props);
+    overrideProps: {
+      as: (wrapper && (wrapper as any).as) || ElementType,
```

Comment:
```suggestion
      as: (wrapper && (wrapper as BoxProps).as) || ElementType,
```

Not sure, but should work